proposals/2023/05-26_vela_ui_rewrite.md

```diff
@@ -0,0 +1,217 @@
+  - re-using the existing cypress tests as a foundation to reach and maintain coverage
+- [Prettier](https://prettier.io/)
+  - `gofmt` alike, automatic formatting without debate
+- [eslint](https://eslint.org/)
+  - `golangci-lint` alike, automatic linting, recommending best coding practices
+
+<details>
+<summary><b>Why react?</b></summary>
+
+React has been a staple frontend "framework" in 2023. It has wide adoption and has a massive practitioner base. In the [2023 Stackoverflow Survey](https://survey.stackoverflow.co/2023/#section-admired-and-desired-web-frameworks-and-technologies), it continues to earn top a top position in desirability and admiration.
+
+We think that React, among others "frameworks" like Angular, Vue or Svelte fulfills our desire to increase casual contributors in this codebase. The principle reason is the sheer momentum that React has.
+
+React, like most frontend tooling, will enter its halcyon days and the industry will move to new tooling. There is no such thing as future-proof. However, in an effort to be future-resistant we are specifically making choices that will be easy to change around build tooling. We are not entirely leveraging the React core team's ecosystem (Next, CRA, Jest), and instead using a broader community ecosystem (Vite).
+
+</details>
+
+<details>
+<summary><b>Can we achieve the same and better results than Elm with this new approach?</b></summary>
+
+Elm offered a bundled developer experience for webapps. The React-centric ecosystem is much more fragmented, but it lets you pick from a variety of options that are easier to plug in and to swap as needs change.
+
+</details>
+
+We have considered these functional changes:
+
+- new/updated endpoints or url paths
+  - re-evaluating the `/-/secrets` path
+  - re-evaluating paths that shadow potentially valid Organization, Repository, or Secret names
+- new/updated configuration variables (environment, flags, files, etc.)
+  - environment variables for features (e.g. default pull request settings, auto-linkifying in logs, etc)
+
+From an overall perspective, the goal is to:
+
+- Replicate the current user interface (look'n'feel; design) in the technology stack listed above as closely as possible
+- Replicate the current functionality in the technology stack listed above as closely as possible
+
+Consider this redesign and rewrite a `1-to-1.3̅3̅3̅` port rather than a `1-to-1` port.
+
+Further enhancements can and will be made after.
+
+## Implementation
+
+<!--
+This section is intended to explain how the solution will be implemented for the proposal.
+
+NOTE: If there are no current plans for implementation, please leave this section blank.
+-->
+
+- Since the UI is a different repo and there are no co-dependency's the new UI should be a drop-in replacement and in theory should be able to run in parallel to the existing version if necessary.
+
+**Please briefly answer the following questions:**
+
+1. Is this something you plan to implement yourself?
```

Comment:
I would encourage you to provide more details on where you see Target engaging. While you might run with the majority of the rewrite, I still think this will take a lot of time/commitment from Target to help identify what feature-parity is, monitor milestones, engage Accessibility experts, and conduct the validation.

Reply:
I hope the Cargill teams responds to this comment as well.

- Feature parity
  - In my opinion, parity is replicating the existing functionality and aligning closely with the look'n'feel
  - Do cypress tests pass with the appropriate changes?
- Monitor milestones
  - Cargill internal dogfooding prior to a GA-release
  - In a later section, there is more discussion on remaining and perhaps details on timelines
  - This feels like a low touch commitment; nobody wants a drawn out parallel implementation indefinitely
- Engage Accessibility experts
  - I am not an accessibility expert, but where it was in place in the existing user interface, it is intended to be replicated, and where it was not implemented already, it will be added where possible
  - I appreciate any expertise and time given towards this; and Cargill has UX experts internally to tap too if required
- Conduct the validation
  - re: dogfooding
  - I appreciate the diligience this could take. Hopefully an equvilent cypress test suite and dogfooding could accommodate part of the validation process
  - There are a dozen pages and few dozen notable page states that I think we could quickly cover

After that, it is TypeScript & React code. The community is broad and could contribute as they need.

Reply:
> Do cypress tests pass with the appropriate changes?
Has anyone looked at the current cypress tests? Are they extensive or should they be reviewed/updated?

> Engage Accessibility experts
Does Cargill have an Accessibility team? I know Target does. I don't know enough of the history of Vela to know if the current implementation went through a review but I strongly believe that any rewrite or mass-revision shoudl include this review (so we don't end up rewriting the rewrite).